src/ol/array.js

```diff
@@ -127,3 +164,153 @@ ol.array.reverseSubArray = function(arr, begin, end) {
     --end;
   }
 };
+
+
+/**
+ * @param {Array.<*>} arr Array.
+ * @return {!Array<?>} Flattened Array.
+ */
+ol.array.flatten = function(arr) {
+  var data = arr.reduce(function(flattened, value) {
+    if (goog.isArray(value)) {
+      return flattened.concat(ol.array.flatten(value));
+    } else {
+      return flattened.concat(value);
+    }
+  }, []);
+  return data;
+};
+
+
+// TODO: Optimisation by storing length or iterating backwards etc
+/**
+ * @param {Array<VALUE>} arr  The array to modify.
+ * @param {Array<VALUE>|VALUE} data The elements or arrays of elements
+ *     to add to arr1.
+ * @template VALUE
+ */
+ol.array.extend = function(arr, data) {
+  var i;
+  var extension = goog.isArrayLike(data) ? data : [data];
+  for (i = 0; i < extension.length; i++) {
+    arr[arr.length] = extension[i];
+  }
+}
+
+
+/**
+ * @param {Array<VALUE>} arr  The array to modify.
+ * @param {VALUE} obj The element to remove.
+ * @template VALUE
+ * @return {boolean} If the element was removed.
+ */
+ol.array.remove = function(arr, obj) {
+  var i = arr.indexOf(obj);
+  if (i > -1) {
+    arr.splice(i, 1);
+  }
+  return i > -1;
+}
+
+
+/**
+ * @param {Array<VALUE>} arr  The array to modify.
+ * @param {?function(this:THISVAL, VALUE, number, ?) : boolean} func The function to compare.
+ * @param {THISVAL=} opt_thisArg Optional this argument for the function.
+ * @template VALUE,THISVAL
+ * @return {VALUE} If the element was removed.
+ */
+ol.array.find = function(arr, func, opt_thisArg) {
+  if (typeof func !== 'function') {
+    throw new TypeError('func must be a function');
```

Comment:
You should change the param type to `{function(this:THISVAL, VALUE, number, ?) : boolean}` and remove the `typeof` check and `throw`.